lib/stat_tracker.rb

```diff
@@ -359,4 +359,49 @@ def lowest_scoring_home_team
       end
     end
   end
+
+  # def get_season(provided_game_id)
+  #   @games.each do |game|
+  #     season = game[:season]
+  #     game_id = game[:game_id]
+  #     if game_id == (provided_game_id)
+  #       return season
+  #     end
+  #   end
+  # end
+
+  def winningest_coach(season)
+    num_of_win = Hash.new(0)
+    @game_teams.each do |game_team|
+      #refactor: use start_with? per Jillian
+      if season[0,4] == game_team.game_id[0,4] && game_team.result == "WIN"
+        num_of_win[game_team.head_coach] += 1
+      end
+    end
+    num_of_win.each do |coach, wins|
+      count_of_games_by_season.each do |s|
+        if s[0] == season
+          num_of_win[coach] = (wins / s[1].to_f * 100).round(2)
+        end
+      end
+    end
+    return winning_coach = num_of_win.key(num_of_win.values.max)
+  end
+  def worst_coach(season)
+    num_of_loss = Hash.new(0)
+    @game_teams.each do |game_team|
+      #refactor: use start_with? per Jillian
+      if season[0,4] == game_team.game_id[0,4] && game_team.result == "LOSS"
+        num_of_loss[game_team.head_coach] += 1
+      end
+    end
+    num_of_loss.each do |coach, loss|
+      count_of_games_by_season.each do |s|
```

Comment:
It's good practice to use descriptive variable names instead of single letters to clarify the code's intention.